.github/workflows/rerun-failed.yml

```diff
@@ -0,0 +1,25 @@
+name: Rerun failed
+
+on:
+  workflow_run:
+    workflows: [Run]
+    types: [completed]
+
+jobs:
+  on-failure:
+    runs-on: ubuntu-22.04
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - name: Check if OpenSearch failed to start
+        id: check-opensearch
+        run: |
+          jobs=$(gh api ${{ github.event.workflow_run.jobs_url }})
+          failed=$(echo "${jobs}" | jq '.jobs | any(.steps | any(.name == "Wait for open search to become ready" and .conclusion == "failure"))')
+          echo "failed=${failed}" >> "$GITHUB_OUTPUT"
```

Comment:
We don't just want to restart any run, but only those where opensearch failed to start